sw/device/tests/chip_power_sleep_load.c

```diff
@@ -0,0 +1,377 @@
+
+static dif_aon_timer_t aon_timer;
+static dif_rv_core_ibex_t rv_core_ibex;
+static dif_pwrmgr_t pwrmgr;
+static dif_rv_timer_t rv_timer;
+static dif_alert_handler_t alert_handler;
+static dif_pwm_t pwm;
+static dif_pinmux_t pinmux;
+static dif_otp_ctrl_t otp_ctrl;
+static dif_gpio_t gpio;
+
+static volatile const bool kCoreClkOff = false;
+static volatile const bool kIoClkOff = false;
+static volatile const bool kUsbSlpOff = false;
+static volatile const bool kUsbActOff = false;
+static volatile const bool kDeepSleep = false;
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// ISRs
+
+void ottf_external_isr(void) {
+  LOG_INFO("SW_INFO: got external IRQ");
+  ext_irq_fired = true;
+}
+
+void ottf_external_nmi_handler(void) {
+  nmi_fired = true;
+
+  expected_isr_handler();
+
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t *)&nmi_state));
+  CHECK_DIF_OK(dif_rv_core_ibex_clear_nmi_state(&rv_core_ibex,
+                                                kDifRvCoreIbexNmiSourceAll));
+}
+
+static void wdog_irq_handler(void) {
+  bool is_pending;
+
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      &aon_timer, kDifAonTimerIrqWdogTimerBark, &is_pending));
+  irq_is_pending = is_pending;
+
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon_timer));
+
+  CHECK_DIF_OK(
+      dif_aon_timer_irq_acknowledge(&aon_timer, kDifAonTimerIrqWdogTimerBark));
+}
+
+void prepare_to_exit(void) {
+  CHECK_DIF_OK(dif_pwm_phase_cntr_set_enabled(&pwm, kDifToggleDisabled));
+
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 0));
+
+  LOG_INFO("SW_INFO: Prepare to exit");
+
+  CHECK(ext_irq_fired == false, "Unexpected external interrupt triggered.");
+
+  pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0);
+
+  aon_timer_testutils_shutdown(&aon_timer);
+
+  test_status_set(kTestStatusInTest);
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rv_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR), &rv_timer));
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+  CHECK_DIF_OK(dif_pwm_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWM_AON_BASE_ADDR), &pwm));
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp_ctrl));
+  CHECK_DIF_OK(
+      dif_gpio_init(mmio_region_from_addr(TOP_EARLGREY_GPIO_BASE_ADDR), &gpio));
+
+  LOG_INFO("SW_INFO: Running CHIP Power Sleep Load test");
+
+  static const uint32_t kGpioMask = 0x00000004;
```

Comment:
should this be static? it is scoped to the `test_main` function already, why do you want its lifetime extended beyond the scope? applies throughout this function

Reply:
This variable is static as we may, in the future, want to override it (from outside the S/W) to use another GPIO pin.
I will make it a global variable to avoid confusion.
Thanks for noticing.

Reply:
done